server/src/application.py

```diff
@@ -0,0 +1,126 @@
+from src.providers import ISODatetimeProvider
+from src.views import views
+
+try:
+    import sentry_sdk
+    from sentry_sdk.integrations.flask import FlaskIntegration
+except ImportError:
+    pass
+
+
+def create_flask_app(config=None):
+    """Create the flask app"""
+    tf_app = APIFlask(__name__)
+    if config:
+        tf_app.config.from_object(config)
+    tf_log = create_logger(tf_app)
+
+    if not tf_app.debug:
+        stream_handler = logging.StreamHandler()
+        stream_handler.setLevel(logging.INFO)
+        tf_log.addHandler(stream_handler)
+
+    tf_app.config["PROPAGATE_EXCEPTIONS"] = True
+
+    # Return datetime objects as RFC3339/ISO8601 strings
+    tf_app.json = ISODatetimeProvider(tf_app)
+
+    if tf_app.config.get("TESTING") is not True:
+        setup_mongodb(tf_app)
+
+    sentry_dsn = tf_app.config.get("SENTRY_DSN")
+    if sentry_dsn and "sentry_sdk" in globals():
+        sentry_sdk.init(  # pylint: disable=abstract-class-instantiated
+            dsn=sentry_dsn, integrations=[FlaskIntegration()]
+        )
+
+    @tf_app.errorhandler(NotFound)
+    def handle_404(exc):
+        tf_log.error("[404] Not found: %s", request.url)
+        return "Not found: {}\n".format(exc), 404
+
+    @tf_app.errorhandler(ConnectionFailure)
+    def handle_timeout(exc):
+        tf_log.exception("pymongo connection failure: %s", exc)
+        return "Server Connection Failure", 500
+
+    @tf_app.errorhandler(Exception)
+    def unhandled_exception(exc):
+        tf_log.exception("Unhandled Exception: %s", (exc))
+        return "Unhandled Exception: {}\n".format(exc), 500
+
+    tf_app.register_blueprint(views)
+    tf_app.register_blueprint(v1, url_prefix="/v1")
+
+    return tf_app
+
+
+def setup_mongodb(application):
+    """
+    Setup mongodb connection if we have valid config data
+    Otherwise leave it empty, which means we are probably running unit tests
+    """
+
+    mongo_user = os.environ.get("MONGODB_USERNAME")
+    mongo_pass = os.environ.get("MONGODB_PASSWORD")
+    if mongo_pass:
+        # password might contain special chars
+        mongo_pass = urllib.parse.quote_plus(mongo_pass)
+
+    mongo_db = os.environ.get("MONGODB_DATABASE")
+    mongo_host = os.environ.get("MONGODB_HOST")
+    mongo_port = os.environ.get("MONGODB_PORT", "27017")
+    mongo_uri = os.environ.get("MONGODB_URI")
+    mongo_auth = os.environ.get("MONGODB_AUTH_SOURCE", "admin")
+
+    if not mongo_uri:
+        if not (mongo_host and mongo_db):
+            raise SystemExit("No MongoDB URI configured!")
+        mongo_creds = (
+            f"{mongo_user}:{mongo_pass}@" if mongo_user and mongo_pass else ""
+        )
+        mongo_uri = (
+            f"mongodb://{mongo_creds}{mongo_host}:{mongo_port}/{mongo_db}"
+            f"?authSource={mongo_auth}"
+        )
+
+    mongo.init_app(
+        application,
+        uri=mongo_uri,
+        uuidRepresentation="standard",
+        serverSelectionTimeoutMS=2000,
```

Comment:
What do you think about making this a configurable parameter?

Reply:
we could, but the default is 30 and we had to set it very high because of some PS5 issue IIRC and haven't had to touch it since then. It's not new - just the file is moving to a new name, so that's why you're seeing it now. I think I'd prefer to defer that to a later change if it becomes something we really need to have a knob for adjusting frequently, otherwise we had previously been advised to keep the number of configuration variables limited if possible.